### PR TITLE
Add cards from LEDE-JP, TW01-JP, and QCDB-JP

### PR DESCRIPTION
**Pull request by StupidStudiosN**

### CODE REVIEW

**Comment by pokeboyexn** [comment]
Hi, may I know what is your contact? Not sure if you are in discord or on QQ. Some of the scripts have bugs and require some testing, which we have already tested. Let's sync and prevent duplicated work/efforts.

**Comment by StupidStudiosN** [comment]
> Hi, may I know what is your contact? Not sure if you are in discord or on QQ. Some of the scripts have bugs and require some testing, which we have already tested. Let's sync and prevent duplicated work/efforts.

Sure, it's StupidStudiosN on Discord as well.  Reach out to me through DMs, and I'd be happy to help.

File: scripts/LEDE-JP/c101204005.lua [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+function s.spop(e,tp,eg,ep,ev,re,r,rp)
+	local c=e:GetHandler()
+	if not c:IsRelateToEffect(e) then return end
+	Duel.SpecialSummon(c,0,tp,tp,false,false,POS_FACEUP)
+end
+function s.spcon2(e,tp,eg,ep,ev,re,r,rp)
+	return e:GetHandler():IsReason(REASON_EFFECT)
+end
+function s.spfilter2(c,e,tp)
+	return c:IsCode(id) and c:IsCanBeSpecialSummoned(e,0,tp,false,false) and (c:IsLocation(LOCATION_HAND+LOCATION_GRAVE) or c:IsFaceup())
```
Comment (Crescent0v0) [inline scripts/LEDE-JP/c101204005.lua:74]:
```suggestion
	return c:IsCode(id) and c:IsCanBeSpecialSummoned(e,0,tp,false,false) and (c:IsLocation(LOCATION_HAND+LOCATION_DECK) or c:IsFaceup())

```

File: scripts/LEDE-JP/c101204004.lua [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	e1:SetProperty(EFFECT_FLAG_DELAY)
+	e1:SetCode(EVENT_DESTROYED)
+	e1:SetCountLimit(1,id+o)
+	e1:SetCondition(s.setcon)
+	e1:SetTarget(s.settg)
+	e1:SetOperation(s.setop)
+	c:RegisterEffect(e1)
+end
+function s.srfilter(c)
+	return c:IsCode(101204051) or (aux.IsCodeListed(c,101204051) and c:IsType(TYPE_SPELL+TYPE_TRAP)) and c:IsAbleToHand()
```
Comment (fallenstardust) [inline scripts/LEDE-JP/c101204004.lua:32]: return (c:IsCode(101204051) or aux.IsCodeListed(c,101204051) and c:IsType(TYPE_SPELL+TYPE_TRAP)) and c:IsAbleToHand()

Comment (fallenstardust) [inline scripts/LEDE-JP/c101204004.lua:32]:
```suggestion
	return (c:IsCode(101204051) or aux.IsCodeListed(c,101204051) and c:IsType(TYPE_SPELL+TYPE_TRAP)) and c:IsAbleToHand()

```

Comment (fallenstardust) [inline scripts/LEDE-JP/c101204004.lua:32]:
```suggestion
	return (c:IsCode(101204051) or aux.IsCodeListed(c,101204051) and c:IsType(TYPE_SPELL+TYPE_TRAP)) and c:IsAbleToHand()

```

**Review by Wind2009-Louse — CHANGES_REQUESTED** [review]

File: scripts/TW01-JP/c100211122.lua [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+end
+function s.sumop(e,tp,eg,ep,ev,re,r,rp)
+	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_SUMMON)
+	local g=Duel.SelectMatchingCard(tp,s.filter,tp,LOCATION_HAND,0,1,1,nil)
+	if g:GetCount()>0 then
+		Duel.Summon(tp,g:GetFirst(),true,nil)
+	end
+end
+function s.repfilter(c,tp)
+	return c:IsFaceup() and c:IsControler(tp) and c:IsLocation(LOCATION_MZONE)
```
Comment (Wind2009-Louse) [inline scripts/TW01-JP/c100211122.lua:58]: Should be able to replace the destruction in Spell & Trap Zones

File: scripts/TW01-JP/c100211122.lua [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+		and c:IsSetCard(0xb5) and c:IsReason(REASON_BATTLE+REASON_EFFECT) and not c:IsReason(REASON_REPLACE)
+end
```
Comment (Wind2009-Louse) [inline scripts/TW01-JP/c100211122.lua:60]:
```suggestion
function s.repfilter(c,tp)
	return c:IsFaceup() and c:IsControler(tp) and c:IsOnField()
		and c:IsSetCard(0xb5) and c:IsReason(REASON_BATTLE+REASON_EFFECT) and not c:IsReason(REASON_REPLACE)
end
```

**Pull request merged**